packages/grafana-data/src/types/datasource.ts

```diff
@@ -245,7 +245,10 @@ abstract class DataSourceApi<
   /**
    * Query for data, and optionally stream results
    */
-  abstract query(request: DataQueryRequest<TQuery>): Promise<DataQueryResponse> | Observable<DataQueryResponse>;
+  abstract query(
+    request: DataQueryRequest<TQuery>,
+    interpolate?: InterpolateFunction
```

Comment:
I think here it should not be optional, implementations should be able to rely on it. But dam that does not work as this function is called a lot FROM plugins, so we have no way to enforce that. 

But having it optional is so messy from the plugin implementation perspective, having to use this but then fallback to something else if it's not there. I think we might need creating brainstorming around this :)

Reply:
Yes - definitelly, just posting this as a possible (quite messy) way to solve the problem.

Let's meet on Monday and talk alternatives through. Tried monkey-patching the templateSrv, but that seems very unreliable.